bluez/tools/hciconfig.c

```diff
@@ -1844,12 +1879,19 @@ static void print_dev_hdr(struct hci_dev_info *di)
 
 	ba2str(&di->bdaddr, addr);
 
+	busname = ((di->type & 0x0f) <= HCI_MAX_BUS)
+		? hci_bustostr(di->type & 0x0f)
```

Comment:
When mixing allocated and non-allocated stings I tend to do something like this:
```C
    char* allocated_string = NULL;
    const char* string_to_print;

    if (need to allocate) {
        string_to_print = allocated_string = allocate_string();
    } else {
        string_to_print = get_static_string();
    }

    printf("here is is: %s\n", string_to_print);

    free(allocated_string);
```
This makes it less likely to forget to free allocated string or accidentally try to deallocate the static one. But that's a matter of taste of course.